### ISSUE TITLE
Add uninstall target/script

### ISSUE THREAD

**Issue by zbeekman** [sourceryinstitute/opencoarrays#89]
Add an uninstall target to CMakeLists.txt and some means of invoking it via a script. (i.e., either by passing a flag like `--remove` or `--uninstall` to `install.sh` which would then use the CMake `uninstall` target or providing a separate script to utilize the CMake target.

**Comment by zbeekman** [comment]
Also, see https://github.com/jacobwilliams/JSON-Fortran/blob/master/CMakeLists.txt for an example of how to do this with CMake.

**Comment by rouson** [comment]
I prefer that there be a separate uninstall.sh script at the top-level of the OpenCoarrays source.  The name "uninstall.sh" immediately communicates the script's purpose and language family (a bourne shell descendent).  By contrast, the "./install.sh --uninstall" would likely seem counterintuitive to most users and, at a minimum, they would probably have to search our documentation to find it.  Nonetheless, it might be reasonable to either have "./install.sh --uninstall" simply invoke "uninstall.sh" or vice versa.  On a related note, at some point, install.sh should be modified to clean up after itself and delete temporary files before exiting.  For now, install.sh takes an extremely conservative approach and deletes (almost)  nothing based on a philosophy that deleting (almost) nothing means never deleting the wrong thing.

**Comment by zbeekman** [comment]
yes, I think you're right, having an `uninstall.sh` is the best way to go. A flag to install.sh that triggers a call to uninstall.sh also sounds good.

**Comment by zbeekman** [comment]
At some point it would be cool to host scripts that drive the entire download and installation, as well as uninstallation as stand-alone entities. I'll have to think a little bit about elegant ways to do this.

**Comment by zbeekman** [comment]
Perhaps as part of #79 I'll try to get `find_package(OpenCoarrays)` up and running, and then we could use CMake in script mode locate the OpenCoarrays installation and nuke it.

**Comment by zbeekman** [comment]
There is a build target for `uninstall` that is generated by CMake upon installation. It would be nice to install a manifest of files (and their hashes, ideally) that are installed with OC, and then have a script that just nukes those files.
